### PR TITLE
Add force store comment

### PR DESCRIPTION
**Pull request by ivadym**
## Description

Context:
- https://github.com/Clinical-Genomics/cg/issues/2702#issuecomment-1822890325

### Added
- Force store comment
- Prevent automatic uploads of force stored cases

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test
- [x] cg workflow <workflow> <case-id> store --force --comment 

### Expected test outcome

- [x] Analysis can be stored
  - [x] Analysis stored with the provided comment
  - [x] Analysis can be uploaded and delivered

  Screenshot 2024-07-31 at 12 10 40

- [x] Delivery report includes the analysis comment

### Misc
- [x] Cleaning of analysis
  - Not affected. `get_analyses_to_clean` uses `started_at`.
- [x] Scout load config file using `completed_at`.
  - Updated to use `started_at` within `add_common_info_to_load_config` when `completed_at` empty.
- [x] Part of the string representation of the`Analysis(Base)` model.
  - Irrelevant if we have `None` as the `completed_at` field

### Filters
- [x] `AnalysisFilter.VALID_IN_PRODUCTION` (`filter_valid_analyses_in_production`)
  - Used in `get_analyses_to_upload`. It will keep having the expected behaviour (we don't want this cases to be automatically uploaded)
  - `analyses_to_delivery_report`: not affected as we'll generate the reports manually
  - `analyses_to_upload_delivery_reports`: not affected as we'll generate the reports manually
- [x] `AnalysisFilter.COMPLETED` (`filter_completed_analyses`)
  - Used in `get_analyses_to_upload`. It will keep having the expected behaviour (we don't want this cases to be automatically uploaded)
- [x] `AnalysisFilter.ORDER_BY_COMPLETED_AT` (`order_analyses_by_completed_at_asc`)
    - Used in `get_analyses_to_upload`, `analyses_to_delivery_report`, `analyses_to_upload_delivery_reports`. Not affected as these analysis won't be picked up for automatic uploads.
- [x] `CaseFilter.NOT_ANALYSED` (`filter_cases_not_analysed`)
  - Used in `get_not_analysed_cases_by_sample_internal_id`/`add_external_data_to_housekeeper`/`cg add external-hk`. Not relevant for this manual command which uses the ticket as an argument.
- [x] `Case.latest_analyzed`
  - Used to extract `cases_to_analyse` and by `is_case_ready_for_analysis`. To retrieve these cases it also takes into account the CaseActions ( `CaseActions.ANALYZE`) which should be empty after an analysis is forcedly stored. Therefore, the behaviour  should not be affected.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/workflow/analysis.py [hunk 1]
```diff
@@ -250,9 +252,10 @@ def upload_bundle_statusdb(self, case_id: str, dry_run: bool = False) -> None:
         new_analysis: Case = self.status_db.add_analysis(
```
Comment (diitaz93) [inline cg/meta/workflow/analysis.py:252]: Should this return an Analysis?
```suggestion
        new_analysis: Analysis = self.status_db.add_analysis(
```

**Review by diitaz93 — APPROVED** [review]
Great 🌟

File: cg/cli/workflow/balsamic/base.py [hunk 2]
```diff
@@ -161,11 +166,13 @@ def store_housekeeper(context: CGConfig, case_id: str, force: bool):
...
+        analysis_api.verify_case_config_file_exists(case_id=case_id, dry_run=dry_run)
         analysis_api.verify_deliverables_file_exists(case_id=case_id)
-        analysis_api.upload_bundle_housekeeper(case_id=case_id, force=force)
-        analysis_api.upload_bundle_statusdb(case_id=case_id)
-        analysis_api.set_statusdb_action(case_id=case_id, action=None)
+        analysis_api.upload_bundle_housekeeper(case_id=case_id, dry_run=dry_run, force=force)
+        analysis_api.upload_bundle_statusdb(
+            case_id=case_id, comment=comment, dry_run=dry_run, force=force
+        )
+        analysis_api.set_statusdb_action(case_id=case_id, action=None, dry_run=dry_run)
```
Comment (islean) [inline cg/cli/workflow/balsamic/base.py:175]: Probably out of scope for this PR but it would be nice to not have all this logic in the CLI file.

Comment (ivadym) [reply cg/cli/workflow/balsamic/base.py:175]: Yes I totally agree, but I'm not sure it's worth the effort giving that we'll try to deprecate Balsamic in the next 6-12 months 😓

**Comment by ivadym** [comment]
Deployed!
Screenshot 2024-08-06 at 09 50 10

**Pull request merged**